compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -843,7 +924,16 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
 
     // Apply inliner to `rhsToInline`, split off any implicit bindings from result, and
     // make them part of `bindingsBuf`. The expansion is then the tree that remains.
-    val expansion = inliner.transform(rhsToInline)
+    val expansion0 = inliner.transform(rhsToInline)
+    val expansion =
+      if opaqueProxies.nonEmpty && !inlinedMethod.is(Transparent) then
```

Comment:
I am not sure what you propose. Right now, we don't do it for Transparent since we return whatever the inlined function returns. We can't cast to the call type by definition. And for normal inlines we already wrap with the call type. The cast just makes sure that that wrapping is always successful.

Reply:
After discussing I understand now. I updated the test to also do transparent inlines, and indeed we need a cast in that case as well.